File: contests/cf1697/B.cpp
#include <vector>
#include <iostream>
#include <algorithm>
#include <numeric>
#include <functional>
using namespace std;

int main(){
    int n, q, t, x, y;
    cin >> n >> q;
    vector<int> price;
    for (int i=0; i<n ; i++){
        cin >> t;
        price.push_back(t);
    }
    sort(price.begin(), price.end(), greater<int>());

    for (int i=0; i<q; i++){
        cin >> x >> y;
        cout << accumulate(price.begin()+(x-y), price.begin()+x, 0);
    }
    return 0;
}

File: contests/cf1697/B.py
from itertools import accumulate
def main():
    n, q = map(int, input().split())
    price = [int(i) for i in input().split()]
    price.sort(reverse=True)
    pref = [0] + list(accumulate(price))
    # print(pref)
    for _ in range(q):
        x, y = map(int, input().split())

        # print(f'{price[x-y:x] = }')
        print(pref[x] - pref[x-y])

    return


if __name__ == "__main__":
    main()
